Mathlib/RingTheory/RingHom/FaithfullyFlat.lean

```diff
@@ -61,6 +61,12 @@ lemma of_bijective (hf : Function.Bijective f) : f.FaithfullyFlat := by
     exact (RingEquiv.ofBijective f hf).injective (by simp)
   rw [← PrimeSpectrum.specComap_comp_apply, this, PrimeSpectrum.specComap_id]
 
+lemma injective (hf : f.FaithfullyFlat) : Function.Injective ⇑f := by
+  algebraize [f]
+  intro _ _ h
+  apply (smul_left_injective' : Function.Injective (· • · : R → S → S))
+  simp only [Algebra.smul_def, RingHom.algebraMap_toAlgebra, h]
```

Comment:
There is also [`FaithfulSMul.algebraMap_injective`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Basic.html#FaithfulSMul.algebraMap_injective)